addons/common/embedded.jsx

```diff
@@ -0,0 +1,62 @@
+/**
+ * Embedded
+ * A reuseable embedded content element. For usage, see the YouTube addon.
+ */
+
+const Field = require('./field')
+const Frame = require('./frame')
+const Graphic = require('./graphic')
+const React = require('react')
+
+const Embedded = React.createClass({
+
+  propTypes: {
+    baseUrl: React.PropTypes.string,
+    onChange: React.PropTypes.func.isRequired,
+    name: React.PropTypes.string.isRequired,
+    resolveUrl: React.PropTypes.func,
+    title: React.PropTypes.string
+  },
+
+  getDefaultProps() {
+    return {
+      baseUrl: "",
+      slug: '',
+      title: "Embedded Content",
+      resolveUrl: (base, slug) => base + slug
+    }
+  },
+
+  getSrc() {
+    const { baseUrl, resolveUrl, slug } = this.props
+
+    return this.hasSlug() ? resolveUrl(baseUrl, slug) : null
+  },
+
+  hasSlug() {
+    return `${ this.props.slug || '' }`.trim().length > 0
```

Comment:
Good thought. If the slug is a number, it would error out.

Having said that, this won't cover cases where the slug is 0. I'll write up a test case and verify coverage.